pytorch_lightning/trainer/trainer.py

```diff
@@ -486,27 +493,37 @@ def __init__(
         self.row_log_interval = row_log_interval
 
         # how much of the data to use
-        # TODO: remove in 1.0.0
-        if overfit_pct > 0:
+        # TODO: remove in 0.10.0
+        if overfit_pct is not None:
+            rank_zero_warn("Argument `overfit_pct` is now set by `overfit_batches` since v0.8.0"
+                           " and this argument will be removed in v0.10.0", DeprecationWarning)
             overfit_batches = overfit_pct
 
         # convert floats to ints
-        overfit_batches = int(overfit_batches) if overfit_batches > 1.0 else overfit_batches
-        self.overfit_batches = overfit_batches
+        self.overfit_batches = _determine_limit_batches(overfit_batches)
 
-        # TODO: remove in 1.0.0
-        if val_percent_check < 1.0:
+        # TODO: remove in 0.10.0
+        if val_percent_check is not None:
+            rank_zero_warn("Argument `val_percent_check` is now set by `limit_val_batches` since v0.8.0"
+                           " and this argument will be removed in v0.10.0", DeprecationWarning)
             limit_val_batches = val_percent_check
 
-        if test_percent_check < 1.0:
+        # TODO: remove in 0.10.0
+        if test_percent_check is not None:
+            rank_zero_warn("Argument `test_percent_check` is now set by `limit_test_batches` since v0.8.0"
+                           " and this argument will be removed in v0.10.0", DeprecationWarning)
             limit_test_batches = test_percent_check
 
-        limit_test_batches = int(limit_test_batches) if limit_test_batches > 1.0 else limit_test_batches
-        limit_val_batches = int(limit_val_batches) if limit_val_batches > 1.0 else limit_val_batches
+        # TODO: remove in 0.10.0
+        if train_percent_check is not None:
+            rank_zero_warn("Argument `train_percent_check` is now set by `limit_train_batches` since v0.8.0"
+                           " and this argument will be removed in v0.10.0", DeprecationWarning)
+            limit_train_batches = train_percent_check
 
-        # TODO: convert train_percent_check to limit_train_batches
-        self.determine_data_use_amount(train_percent_check, limit_val_batches,
-                                       limit_test_batches, overfit_batches)
+        self.limit_test_batches = _determine_limit_batches(limit_test_batches)
```

Comment:
this is unecessary and hides a very simple op. Please revert

Reply:
it prevents typo as it is used several time in the code